pkg/component/runtime/state.go

```diff
@@ -345,6 +347,10 @@ func (s *ComponentState) syncCheckin(checkin *proto.CheckinObserved) bool {
 			s.VersionInfo.Name = checkin.VersionInfo.Name
 			changed = true
 		}
+		if checkin.VersionInfo.BuildHash != "" && s.VersionInfo.BuildHash != checkin.VersionInfo.BuildHash {
+			s.VersionInfo.BuildHash = checkin.VersionInfo.BuildHash
+			changed = true
+		}
```

Comment:
If we are trying to detect a change in the version reported by the component we should check the triplet (version, snapshot flag, hash) since repackaged agent versions will have the same hash as other versions.

If that's not what we are doing here, then I don't understand what purpose this snippet serves :sweat_smile:

Reply:
it's the components build hash. the agent does not have it bu default. the component sends it on its 1st checkin

Reply:
Yes this is the build hash of the sub-processes, which are not going to match the build hash of the agent. 

We stopped receiving the version string because it is now just going to be equal to the package version of the agent. 

Looking at only the hash is intentionally, because it is the only thing that tells us which commit and branch the component version is from, since the version string no longer uniquely identifies a stack release branch.